### ISSUE TITLE
URIs with escaped opening curly braces are processed incorrectly

### ISSUE THREAD

**Issue by alkedr** [yandex-qatools/uri-differ#8]
`param("z").apply(URI.create("/?y=%7B%22"));` (%7B is `{`, %22 is `"`)

throws

```java
	java.lang.IllegalArgumentException: Illegal character """ at position 3 is not allowed as a start of a name in a path template "y={"&z=(ignored)".
	at org.glassfish.jersey.uri.internal.UriTemplateParser.parseName(UriTemplateParser.java:328)
	at org.glassfish.jersey.uri.internal.UriTemplateParser.parse(UriTemplateParser.java:251)
	at org.glassfish.jersey.uri.internal.UriTemplateParser.<init>(UriTemplateParser.java:110)
	at org.glassfish.jersey.uri.UriTemplate.createUriComponent(UriTemplate.java:1001)
	at org.glassfish.jersey.uri.UriTemplate.createURIWithStringValues(UriTemplate.java:967)
	at org.glassfish.jersey.uri.UriTemplate.createURIWithStringValues(UriTemplate.java:906)
	at org.glassfish.jersey.uri.UriTemplate.createURI(UriTemplate.java:871)
	at org.glassfish.jersey.uri.internal.JerseyUriBuilder._build(JerseyUriBuilder.java:893)
	at org.glassfish.jersey.uri.internal.JerseyUriBuilder.build(JerseyUriBuilder.java:810)
	at ru.lanwen.diff.uri.core.filters.AnyParamValueFilter.apply(AnyParamValueFilter.java:50)
	at ru.yandex.autotests.market.report.backend.tests.redirect.ATest.name(ATest.java:14)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:160)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:234)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:74)
```

I guess this is because `javax.ws.rs.core.UriBuilder.build(Object... values)` tries to replace things like `{a}` with elements of array that it takes as parameter.

Is there an easy way to fix this?

**Comment by lanwen** [comment]
hmm, it works fine if i use encoding twice:
```java
String encode = "/?y=" +  URLEncoder.encode(URLEncoder.encode("{\"", "UTF-8"), "UTF-8");
param("z").apply(URI.create(encode));
```

**Comment by lanwen** [comment]
Thanks for the report!
Will fix that soon. Found that with filter query is decoded twice.

**Comment by alkedr** [comment]
Thanks!
Could you also update uri-differ-lib version in [matchers-java](https://github.com/yandex-qatools/matchers-java)?

**Comment by lanwen** [comment]
released as 1.2, will be available soon
